### PR TITLE
fix(parser): restore table option INSERT_METHOD should use WriteKeyWord

### LINKED ISSUE #38368: parser: table option should always restore without quotes
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)
```go
package test
import (
	_ "github.com/pingcap/tidb/types/parser_driver"
	"github.com/stretchr/testify/require"
	"github.com/pingcap/tidb/parser"
	"github.com/pingcap/tidb/parser/ast"
	"github.com/pingcap/tidb/parser/format"
	"github.com/pingcap/tidb/parser/model"
)
func TestTableOption(t *testing.T) {
	tests := []struct {
		new  string
		want string
	}{
		{
			new:  `CREATE TABLE book(id INT) INSERT_METHOD = FIRST;`,
			want: "ALTER TABLE `book` INSERT_METHOD = FIRST;\n",
		},
	}
	a := require.New(t)
	
	for _, test := range tests {
		nodes := parser.New().Parse(test.new, "", "")
		get := deparse(diff, format.DefaultRestoreFlags|format.RestoreStringWithoutCharset)
		a.Equal(test.want, get)
	}
}

func deparse(nodes []ast.Node, flag format.RestoreFlags) (string, error) {
	var buf bytes.Buffer
	for _, node := range nodes {
		if err := node.Restore(format.NewRestoreCtx(flag, &buf)); err != nil {
			return "", err
		}
		if _, err := buf.Write([]byte(";\n")); err != nil {
			return "", err
		}
	}
	return buf.String(), nil
}
```
We can find that the `get` is:
```SQL
ALTER TABLE `book` INSERT_METHOD = 'FIRST';\n
```
It's caused by `ctx.WriteString(n.StrValue)` in [here](https://sourcegraph.com/github.com/pingcap/tidb/-/blob/parser/ast/ddl.go?L2301). We will encounter the error if we execute it in MySQL client:
image

### 2. What did you expect to see? (Required)
Restore should restore INSERT_METHOD without quotes likes:
```SQL
CREATE TABLE `tbl`(id INT) INSERT_METHOD = FIRST.
```
https://dev.mysql.com/doc/refman/8.0/en/create-table.html
CleanShot 2022-10-10 at 17 59 18

### 3. What did you see instead (Required)
```SQL
CREATE TABLE `tbl`(id INT) INSERT_METHOD = 'FIRST'.
```
### 4. What is your TiDB version? (Required)
Parser at HEAD.

### PR DESCRIPTION
**Pull request by h3n4l**
### What problem does this PR solve?

Problem Summary:
Issue Number: close #38368
Restore table option `INSERT_METHOD` should always be without quotes.
CleanShot 2022-10-10 at 14 12 25@2x

### What is changed and how it works?
Use `WriteKeyWord()` instead of `WriteString()`, both upper case and lower case can works well.
CleanShot 2022-10-10 at 14 13 18@2x

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: parser/parser_test.go [hunk 1]
```diff
@@ -3629,6 +3629,9 @@ func TestDDL(t *testing.T) {
 		{"ALTER TABLE t STATS_OPTIONS=default", true, "ALTER TABLE `t` STATS_OPTIONS=DEFAULT"},
 		{"ALTER TABLE t STATS_OPTIONS=DeFaUlT", true, "ALTER TABLE `t` STATS_OPTIONS=DEFAULT"},
 		{"ALTER TABLE t STATS_OPTIONS", false, ""},
+
+		// Restore INSERT_METHOD table option
+		{"create table t (a int) INSERT_METHOD=FIRST", true, "CREATE TABLE `t` (`a` INT) INSERT_METHOD = FIRST"},
```
Comment (JmPotato) [inline parser/parser_test.go:3634]: Better follow others' styles to use the upper case for key words.

```suggestion
		{"CREATE TABLE t (a int) INSERT_METHOD=FIRST", true, "CREATE TABLE `t` (`a` INT) INSERT_METHOD = FIRST"},
```

**Review by JmPotato — COMMENTED** [review]
Rest LGTM.

File: parser/parser_test.go (hunk 1 above)
Comment (h3n4l) [reply parser/parser_test.go:3634]: Thx~ : )

**Review by JmPotato — APPROVED** [review]

**Review by xhebox — APPROVED** [review]

**Comment by xhebox** [comment]
@h3n4l Please open a corresponding bug issue. I'll merge this PR after that.

**Comment by h3n4l** [comment]
> @h3n4l Please open a corresponding bug issue. I'll merge this PR after that.

Hey @xhebox , I submit an issue #38368 related to this bug.

**Pull request merged**